site.yml

```diff
@@ -1,4 +1,48 @@
 ---
-- include: redhat_release.yml
-- include: autotested.yml
-- include: cleanup.yml
+
+# Setup/override global variables
+- hosts: all
+  tasks:
+    # Defines adept_job_no_group and rt_adept_job_no
+    - include: roles/common/tasks/adept_job_no.yml
+
+# Everything else depends on slave setup, run cleanup last
+- hosts: localhost
+  roles:
+    - role: slave
+      when: adept_context != 'cleanup'
+  tasks:
+    - debug: var=adept_job_no_group
+
+- hosts: peons
+  roles:
+    - role: peon
+      # During cloud-type cleanup, peons will be deleted
+      when: adept_context == "cleanup"
+      # In case other jobs are running concurrently
+      with_items: "{{ groups[adept_job_no_group] }}"
```

Comment:
ohh, that reminds me, I should probably not use ``item`` as the loop variable (it can clash with deeper/nested loops)...